core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -355,17 +378,22 @@ class DynamicBrokerConfig(private val kafkaConfig: KafkaConfig) extends Logging
     props
   }
 
-  private[server] def validate(props: Properties, perBrokerConfig: Boolean): Unit = CoreUtils.inReadLock(lock) {
-    validateConfigs(props, perBrokerConfig)
+  private def validatedKafkaProps(propsOverride: Properties, perBrokerConfig: Boolean): Map[String, String] = {
```

Comment:
If the lock is still needed here, we should add a comment to that effect

Reply:
Yes, added comment, thanks.